core/src/main/java/org/apache/druid/java/util/emitter/service/SegmentMetadataEvent.java

```diff
@@ -34,11 +37,29 @@ public class SegmentMetadataEvent implements Event
   public static final String VERSION = "version";
   public static final String IS_COMPACTED = "isCompacted";
 
+  /**
+   * Time at which the segment metadata event is created
+   */
   private final DateTime createdTime;
+  /**
+   * Datasource for which the segment is committed
+   */
   private final String dataSource;
+  /**
+   * Start interval of the committed segment
+   */
   private final DateTime startTime;
+  /**
+   * End interval of the committed segment
+   */
   private final DateTime endTime;
+  /**
+   * Version of the committed segment
+   */
   private final String version;
+  /**
+   * Is the segment, a compacted segment or not
+   */
```

Comment:
there are a bunch of getter for those fields e.g. `getCreatedTime()` that were removed when we submitted the patch upstream, it might make sense to remove those here as well.

Reply:
oh I see, they are used to convert the event to proto, so we might not be able to remove them.
We should file a follow-up to clean up this patch and move the proto-related methods/code into separate classes (and eventually into its own extension) to reduce the risk of having merge conflicts.